test/plug/adapters/test/conn_test.exs

```diff
@@ -59,4 +59,21 @@ defmodule Plug.Adapters.Test.ConnTest do
     assert {:ok, params, _} = adapter.parse_req_multipart(state, 1_000_000, fn _ -> :ok end)
     assert params == %{"a" => "b", "c" => [%{"d" => "e"}, "f"]}
   end
+
+  test "use existing conn.host if exists" do
+    conn_with_host = conn(:get, "http://www.elixir-lang.org/")
+    assert "www.elixir-lang.org" == conn_with_host.host
```

Comment:
Maybe we can do

```elixir
assert conn_woth_host.host == "www.elixir-lang.org"
```

as it reads more familiar (at least to me). What do you think?